src/s3contents.ts

```diff
@@ -634,20 +616,18 @@ export class Drive implements Contents.IDrive {
         await this._s3Client.send(command);
 
       if (Contents) {
-        Contents.forEach(c => {
+        const promises = Contents.map(async c => {
           // delete each file with given path
-          this.delete_file(c.Key!);
+          await this.delete_file(c.Key!);
         });
+        await Promise.all(promises);
```

Comment:
I would suggest:
```ts
if (Contents) {
  await Promise.all(Contents.map(c => this.delete_file(c.Key!)));
}
```
because `delete_file` already returns `Promise<void>` so you don't need to `await` it and you also don't need to wrap it in an `async` function.